test/parallel/test-single-executable-application.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+const common = require('../common');
+
+// This tests the creation of a single executable application.
+
+const tmpdir = require('../common/tmpdir');
+const { copyFileSync, readFileSync, writeFileSync } = require('fs');
+const { execSync } = require('child_process');
+const { join } = require('path');
+const { strictEqual } = require('assert');
+
+if (!process.config.variables.single_executable_application)
+  common.skip('Single Executable Application support has been disabled.');
+
+if (!['darwin', 'win32', 'linux'].includes(process.platform))
+  common.skip(`Unsupported platform ${process.platform}.`);
+
+if (process.config.variables.asan)
+  common.skip('Running the resultant binary fails with `Segmentation fault (core dumped)`.');
```

Comment:
This one feels particularly important :) Is there already a way to track these issues?

Reply:
This doesn't segfault for me locally on my Intel macOS when I configure node with `--enable-asan` but it prints this at start up:
```
malloc: nano zone abandoned due to inability to preallocate reserved vm space.
```
I remember seeing this fail on GitHub actions and it looks like we only test asan builds on Ubuntu, not macOS, so I might have to have add an if-linux to this.

Reply:
@addaleax does ^ address your comment?

Reply:
I think the main question here (and for the one below as well) is “where are these issues being tracked?”, or is it just in the list of tests here?

Reply:
@addaleax We're tracking the issues in https://github.com/nodejs/postject/issues but I haven't creates issues for all of these bugs yet. I would need access to the CI systems and some time to come up with issues that are detailed enough to help track down the bugs. Since most of these bugs are upstream bugs in the http://github.com/lief-project/LIEF repo, I thought it would be more reasonable to create the issues in the LIEF repo once we have a Node.js release with this change, so that it can be easily reproducible.